crates/holochain/src/conductor/api/api_external/admin_interface.rs

```diff
@@ -83,16 +106,39 @@ impl AdminInterfaceApi for RealAdminInterfaceApi {
                 // Install Dnas
                 let tasks = dnas.into_iter().map(|dna_payload| async {
                     let InstallAppDnaPayload {
-                        path,
+                        path: maybe_path,
+                        hash: maybe_hash,
                         properties,
                         membrane_proof,
                         nick,
                     } = dna_payload;
-                    let dna = read_parse_dna(path, properties).await?;
-                    let hash = dna.dna_hash().clone();
-                    let cell_id = CellId::from((hash.clone(), agent_key.clone()));
-                    self.conductor_handle.install_dna(dna).await?;
-                    ConductorApiResult::Ok((InstalledCell::new(cell_id, nick), membrane_proof))
+                    if maybe_path.is_none() && maybe_hash.is_none() {
+                        return Err(ConductorApiError::DnaReadError("Neither path nor hash specified in payload".to_string()))
+                    };
+                    if maybe_path.is_some() && maybe_hash.is_some() {
+                        return Err(ConductorApiError::DnaReadError("Both path and hash specified in payload, pick just one".to_string()))
+                    }
+                    if let Some(path) = maybe_path {
+                        // TODO: this if let will be removed after deprecation period
+                        tracing::info!("specifying dna by path with register side-effect is deprecated, please use RegisterDna and install by hash");
```

Comment:
Could use the [deprecation attribute](https://doc.rust-lang.org/edition-guide/rust-2018/the-compiler/an-attribute-for-deprecation.html) but not sure it really matters

Reply:
It would be good to use deprecation attr so we can more easily track it.

Also this should probably be a `warn`, not an `info`.
```suggestion
                        tracing::warn!("specifying dna by path with register side-effect is deprecated, please use RegisterDna and install by hash");
```